src/components/admin/admin-item.js

```diff
@@ -1,48 +1,27 @@
 adminApp.AdminItem = {
   controller: function(args){
-    this.contribution = args.contribution;
-    this.contribution.user_profile_img = this.contribution.user_profile_img || '/assets/catarse_bootstrap/user.jpg';
-    this.CSSsuccess = '.text-success';
-    this.CSSwaiting = '.text-waiting';
-    this.CSSerror   = '.text-error';
-    this.paymentDetails = function(){
-      this.contribution.gateway = this.contribution.gateway.toLowerCase();
-      if(this.contribution.gateway_data){
-        switch(this.contribution.gateway){
-          case 'moip':
-            this.contribution.card_first_digits = this.contribution.gateway_data.cartao_bin;
-            this.contribution.card_last_digits = this.contribution.gateway_data.cartao_final;
-            this.contribution.card_brand = this.contribution.gateway_data.cartao_bandeira;
-            this.contribution.installments = this.contribution.gateway_data.parcelas;
-            return true;
-          case 'pagarme':
-            this.contribution.card_first_digits = this.contribution.gateway_data.card_first_digits;
-            this.contribution.card_last_digits = this.contribution.gateway_data.card_last_digits;
-            this.contribution.card_brand = this.contribution.gateway_data.card_brand;
-            this.contribution.installments = this.contribution.gateway_data.installments;
-            return true;
-          default:
-            return false;
-        }
-      }
-      else{
-        return false;
-      }
-    };
-    this.stateClass = function(){
-      switch(this.contribution.state){
+    var contribution = args.contribution,
+        userProfile, stateClass, paymentMethodClass, displayDetailBox;
+
+    userProfile = function(){
+      return contribution.user_profile_img || '/assets/catarse_bootstrap/user.jpg';
+    }
+
+    stateClass = function(){
```

Comment:
The point contextualizing the CSS class return here was to avoid changing specs whenever we change style. Perhaps making this switch statement an object would allow us to avoid this problem with specs while also avoiding the use of `this`.